### DISCUSSION TITLE
Battery calibration?

### DISCUSSION THREAD

**Discussion by progzone122** [moto-penangf/discussions#5] [Battery] [answered]
### After erasing the hw partition via mtkclient, the battery percentage quickly increased from 0% to 70%, and now the phone seems to not shut down. This partition likely contains battery calibration settings.

<img src="https://github.com/user-attachments/assets/93f3cf8d-c8e0-4db0-8e54-5cd443b84dd9" width="500">
<img src="https://github.com/user-attachments/assets/a1463425-64f3-4baf-b6b4-a25c37230b45" height="500">

### Confirmation of this is the "battid" line in the dump of this partition.

![изображение](https://github.com/user-attachments/assets/a570ee40-0a28-4c55-8bfe-4f81b917c956)

However, there is an issue with charging: it is happening very slowly. Perhaps it is necessary to charge the battery to the maximum level, then fully discharge it to 0%, and recharge it again?

**Comment by shomykohai** [comment]
In my opinion that's not the calibration data  but *related*

What I mean is that the calibration data is stored inside persist iirc, and maybe it has a reference to the battery id? 

I'm just making a theory though


Also same sequence of 1s and 0s appear in proinfo.img 🤔

**Reply by progzone122** [reply]
> In my opinion that's not the calibration data but _related_
> 
> What I mean is that the calibration data is stored inside persist iirc, and maybe it has a reference to the battery id?
> 
> I'm just making a theory though
> 
> Also same sequence of 1s and 0s appear in proinfo.img 🤔

Maybe erase persist helped too, because I cleared these two partitions together

**Reply by progzone122** [reply]
If this actually helps me get my device working again, I'll add it to the documentation

**Reply by shomykohai** [reply]
Welp, check your sensors then, you might have erased calibration data for them too (gyroscope for example)

**Reply by progzone122** [reply]
The persist and hw are generated automatically (if they are missing), so nothing was reset (but of course, I made a backup of the previous version), and all sensors are working fine.

**Reply by progzone122** [reply]
The only thing that bothers me is that the phone charges very slowly - I hope that this will only be the first cycle of charging and after that the charging will work normally

**Reply by progzone122** [reply]
The idea to clear the hw partition came to me after I saw posts on the forum about the Mi A2 Lite phone, where this method was used to fix the proximity sensor.

**Reply by shomykohai** [reply]
> The idea to clear the hw partition came to me after I saw posts on the forum about the Mi A2 Lite phone, where this method was used to fix the proximity sensor.

If you confirm that after some battery cycles it will work normally again, then definitely add it on the documentation as a solution.

Then to be sure that every sensor work right, you should also check the test menu for correct function of *all* sensor (even less important one)

**Reply by progzone122** [reply]
> > The idea to clear the hw partition came to me after I saw posts on the forum about the Mi A2 Lite phone, where this method was used to fix the proximity sensor.
> 
> If you confirm that after some battery cycles it will work normally again, then definitely add it on the documentation as a solution.
> 
> Then to be sure that every sensor work right, you should also check the test menu for correct function of _all_ sensor (even less important one)

Absolutely! We are very lucky that we have access to these partitions via mtkclient - otherwise the device could be thrown away

**Reply by progzone122** [reply]
At least for me it doesn't turn off at 75% now (it used to turn off at 45%, WTF?????)

**Reply by shomykohai** [reply]
> We are very lucky that we have access to these partitions via mtkclient

We should make a list of partitions in which it can be written into, this would help a lot for testing stuff

**Comment by progzone122** [comment]
After the battery was discharged by 5%, the phone turned off again (the battery charge before turning off showed 68%) and after that, when I tried to turn it on, it showed 0% charge. 
I used MTKClient and run command:
```shell
mtk reset
```
After rebooting, it showed 95% charge. WTFF???

**Reply by progzone122** [reply]
I will continue to turn it on and drain the battery using this method, it looks like a software problem

**Comment by progzone122** [comment, marked as answer]
Looks like the BMS in the battery died

**Reply by shomykohai** [reply]
Well this is good news! The phone is still working then!

**Reply by progzone122** [reply]
> Well this is good news! The phone is still working then!

But I'm still not 100% sure. I need a new battery

**Reply by shomykohai** [reply]
> > Well this is good news! The phone is still working then!
> 
> But I'm still not 100% sure. I need a new battery

I might need one too in the future, I can feel the shape of the battery from the backcover, it's probably swelling

If we manage to unlock bl, I'll buy a new battery, otherwise I'll just get a decent phone that provides BL unlocking without code surely

Funny enough is that they don't even seem to care on the forum, I just found out the posts and the stuff I've read are concerning. What happens if the battery decides to go on fire randomly? Probably will only get told that the device support has ended so they can't do anything about it, even if it's an hardware issue from them

**Reply by progzone122** [reply]
> > > > Well this is good news! The phone is still working then!
> > > 
> > > 
> > > But I'm still not 100% sure. I need a new battery
> > 
> > 
> > I might need one too in the future, I can feel the shape of the battery from the backcover, it's probably swelling
> > If we manage to unlock bl, I'll buy a new battery, otherwise I'll just get a decent phone that provides BL unlocking without code surely
> > Funny enough is that they don't even seem to care on the forum, I just found out the posts and the stuff I've read are concerning. What happens if the battery decides to go on fire randomly? Probably will only get told that the device support has ended so they can't do anything about it, even if it's an hardware issue from them

The forum is made up of bots, we spammed repeatedly with questions on the forum but now they just ignore us. Trash Brand.

I'll buy a new battery because it's more cost effective than buying a new phone....
+ I want to continue to help unlock this shit

P.S. I really hope the battery is the problem and the phone has not been damaged in any way

**Reply by progzone122** [reply]
> > > Well this is good news! The phone is still working then!
> > 
> > 
> > But I'm still not 100% sure. I need a new battery
> 
> I might need one too in the future, I can feel the shape of the battery from the backcover, it's probably swelling
> 
> If we manage to unlock bl, I'll buy a new battery, otherwise I'll just get a decent phone that provides BL unlocking without code surely
> 
> Funny enough is that they don't even seem to care on the forum, I just found out the posts and the stuff I've read are concerning. What happens if the battery decides to go on fire randomly? Probably will only get told that the device support has ended so they can't do anything about it, even if it's an hardware issue from them

We progressed that far yet, but we just need to keep going —  if only for the sake of our principles.

**Reply by shomykohai** [reply]
> The forum is made up of bots, we spammed repeatedly with questions on the forum but now they just ignore us. Trash Brand.

No wonder they don't fix bugs, they don't even care about looking at reports

**Reply by progzone122** [reply]
> > The forum is made up of bots, we spammed repeatedly with questions on the forum but now they just ignore us. Trash Brand.
> 
> No wonder they don't fix bugs, they don't even care about looking at reports

It took them a year and a half to fix the issue with no sound during calls. That's all you need to know about quality control at Motorola

**Reply by shomykohai** [reply]
> P.S. I really hope the battery is the problem and the phone has not been damaged in any way

If you confirmed the battery BMS is dead, there's a high chance its just that, especially considering you are experiencing the same issues as other people with dying batteries (random power offs, reboots, incorrect percentage etc.)

**Reply by shomykohai** [reply]
> > > The forum is made up of bots, we spammed repeatedly with questions on the forum but now they just ignore us. Trash Brand.
> > 
> > No wonder they don't fix bugs, they don't even care about looking at reports
> 
> It took them a year and a half to fix the issue with no sound during calls. That's all you need to know about quality control at Motorola

I just hope they fix the issue of the lockscreen moving before ending support, it's so annoying
